parser/src/parsercontext.cpp

```diff
@@ -0,0 +1,81 @@
+#include <boost/filesystem.hpp>
+
+#include <model/file.h>
+#include <model/file-odb.hxx>
+
+#include <util/hash.h>
+#include <util/odbtransaction.h>
+
+#include <parser/parsercontext.h>
+#include <parser/sourcemanager.h>
+
+namespace po = boost::program_options;
+
+namespace cc
+{
+namespace parser
+{
+
+ParserContext::ParserContext(
+    std::shared_ptr<odb::database> db_,
+    SourceManager& srcMgr_,
+    std::string& compassRoot_,
+    po::variables_map& options_):
+      db(db_),
+      srcMgr(srcMgr_),
+      options(options_),
+      compassRoot(compassRoot_)
+{
+  std::unordered_map<std::string, std::string> fileHashes;
+
+  (util::OdbTransaction(this->db))([&]
+   {
+     // Fetch all files from SourceManager
+     std::vector<model::FilePtr> files = this->srcMgr.getAllFiles();
+     files.erase(std::remove_if(files.begin(), files.end(),
+                                [](const auto &item)
+                                {
+                                  return item->type == model::File::DIRECTORY_TYPE ||
+                                         item->type == model::File::BINARY_TYPE;
+                                }), files.end());
+
+     for (model::FilePtr file : files)
+     {
+       if (boost::filesystem::exists(file->path))
+       {
+         if (!this->fileStatus.count(file->path))
+         {
+           auto content = file->content.load();
+           fileHashes[file->path] = content != nullptr ? content->hash : "";
+           if (content == nullptr)
+             continue;
+
+           std::ifstream fileStream(file->path);
+           std::string fileContent(
+               (std::istreambuf_iterator<char>(fileStream)),
```

Comment:
Vexparse parentheses can be replaced with uniform initialization again:

```cpp
std::string fileContent(
  std::istreambuf_iterator<char>{fileStream},
  std::istreambuf_iterator<char>{});
```

(And indentation is off.)

----
I say this stort of string initialization is **ugly**, because if you mess up the vexparse you don't get compile error, but rather some nice converting constructor and weirdo results.